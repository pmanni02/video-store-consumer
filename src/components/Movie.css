.movie {
  margin: 1.5em;
  padding-left: 1.5em;
  padding-right: 1.5em;
  width: 12em;
  background-color: #EEE5E9;
}

.movie img:hover {
  cursor: pointer;
}

/* *{
  border: black 1px solid;
} */
