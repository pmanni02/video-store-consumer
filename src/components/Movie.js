import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Movie extends Component {

  rentalDetail = () => {
    this.props.movieCallback(this.props.title)
  }

  render() {
    return (
      <article>
        <div>{this.props.title}</div>
        <img src={this.props.image_url} alt={this.props.title}  onClick={this.rentalDetail}/>
        <p>{this.props.overview}</p>
        <p>{this.props.release}</p>
      </article>
    )
  }
}

export default Movie;

Movie.propTypes = {
    id: PropTypes.number,
    title: PropTypes.string,
    overview: PropTypes.string,
    release: PropTypes.string,
    image_url: PropTypes.string,
    movieCallback: PropTypes.func,
}
