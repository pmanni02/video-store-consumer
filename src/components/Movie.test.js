import React from 'react';
import { shallow } from 'enzyme';
import Movie from './Movie';

describe('Movie', () => {
  test('snapshot', () => {
    const wrapper = shallow(
      <Movie
        id={0}
        title=''
        overview=''
        release=''
        image_url=''
        movieCallback={()=>{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
