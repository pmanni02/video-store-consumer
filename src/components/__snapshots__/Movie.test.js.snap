// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie snapshot 1`] = `
<article
  className="movie"
>
  <h3 />
  <img
    alt=""
    onClick={[Function]}
    src=""
    title="Click to select"
  />
  <p />
  <p />
</article>
`;
